CONTRIBUTING.md

```diff
@@ -3,118 +3,232 @@
+6. Open `supertokens-root` in your IDE ([IntelliJ IDEA][intellij] is recommended)
+7. After gradle has imported all the dependencies you can start modifying the code
+
+## Developing With IntelliJ IDEA
+
+### Run Configurations
+![run configurations menu][image-run-configurations]
+
+Several Run Configurations are available to run `supertokens-root` scripts:
+- **Lint:** `./gradlew spotlessApply`
+    - Run this configuration before committing
+- **Run Core:** `./runCore`
+- **Run All Tests:** `./startTestEnv`
+- **Start Testing Environment:** `./startTestEnv --wait`
+    - Run this configuration when running individual tests
+- **Load Modules:** `./loadModules`
+
+### Running All Tests
+1. Make sure that none of these are running:
+    - The `Start Testing Environment` run configuration
+    - The `Run All Tests` run configuration
+    - The `./startTestEnv` script
+2. Run the `Run All Tests` run configuration
+
+### Running Single Tests
+1. Run the `Start Testing Environment` run configuration
+2. Wait for it to print `Test environment started!`
+3. Leave this script running in the terminal
+4. Click the Play button next to the test(s) that you want to run
+5. Tests can be debugged with breakpoints
+
+### Running the Core
+1. Edit `supertokens-root/supertokens-core/devConfig.yaml`
+2. Run the `Run Core` run configuration
+3. Breakpoints are **not supported** at this time
+
+## Developing Without IntelliJ IDEA
+
+### Running All Tests (Manually)
+1. Navigate to the `supertokens-root` repository
+2. Run `./startTestEnv`
+3. If all tests pass the terminal should display
+   - Core Tests:  
+     ![core tests passing][image-core-tests-passing]
+   - Plugin Tests:  
+     ![plugin tests passing][image-plugin-tests-passing]
+
+### Running All Tests (GitHub Actions)
+1. Go to the `supertokens-core` repo on GitHub (or your forked version of it)
+2. Navigate to the Actions tab
+3. Find the action named "Run tests" and navigate to it
+4. Click on the "Run workflow" button
 5. Set the config variables in the drop down:
-    - **supertokens-plugin-interface repo owner name**: If you have forked the supertokens-plugin-interface repo, then set the value of this to your github username.
-    - **supertokens-plugin-interface repos branch name**: If the core version you are working on is compatible with a plugin-interface version that is not in the master branch, then set the correct branch name in this value.
-6. Click on "Run workflow".
+    - **supertokens-plugin-interface repo owner name**: If you have forked the supertokens-plugin-interface repo, then set the value of this to your GitHub username
+    - **supertokens-plugin-interface repos branch name**: If the core version you are working on is compatible with a plugin-interface version that is not in the master branch, then set the correct branch name in this value
+6. Click on "Run workflow"
 
-## Running the core manually
+## Running the Core (Manually)

```

Comment:
```suggestion
## Running the Core (Using Java)

```

Reply:
I have clarified this in this document. It now reads `With Console`.